x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -4,90 +4,178 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { InfraClientCoreSetup } from '../types';
-import { LogsFetchDataResponse } from '../../../observability/public';
+import { DEFAULT_SOURCE_ID } from '../../common/constants';
+import { InfraClientCoreSetup, InfraClientStartDeps } from '../types';
+import {
+  FetchData,
+  LogsFetchDataResponse,
+  HasData,
+  FetchDataParams,
+} from '../../../observability/public';
+import { callFetchLogSourceConfigurationAPI } from '../containers/logs/log_source/api/fetch_log_source_configuration';
+import { callFetchLogSourceStatusAPI } from '../containers/logs/log_source/api/fetch_log_source_status';
 
-export function getLogsHasDataFetcher(getStartServices: InfraClientCoreSetup['getStartServices']) {
-  return async () => {
-    // if you need the data plugin, this is how you get it
-    // const [, startPlugins] = await getStartServices();
-    // const { data } = startPlugins;
+interface StatsAggregation {
+  buckets: Array<{ key: string; doc_count: number }>;
+}
+
+interface SeriesAggregation {
+  buckets: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+    dataset: StatsAggregation;
+  }>;
+}
 
-    // if you need a core dep, we need to pass in more than just getStartServices
+interface LogParams {
+  index: string;
+  timestampField: string;
+}
+
+type StatsAndSeries = Pick<LogsFetchDataResponse, 'stats' | 'series'>;
 
-    // perform query
-    return true;
+export function getLogsHasDataFetcher(
+  getStartServices: InfraClientCoreSetup['getStartServices']
+): HasData {
+  return async () => {
+    const [core] = await getStartServices();
+    const sourceStatus = await callFetchLogSourceStatusAPI(DEFAULT_SOURCE_ID, core.http.fetch);
+    return sourceStatus.data.logIndexNames.length > 0;
   };
 }
 
 export function getLogsOverviewDataFetcher(
   getStartServices: InfraClientCoreSetup['getStartServices']
-) {
-  return async (): Promise<LogsFetchDataResponse> => {
-    // if you need the data plugin, this is how you get it
-    // const [, startPlugins] = await getStartServices();
-    // const { data } = startPlugins;
+): FetchData<LogsFetchDataResponse> {
+  return async (params) => {
+    const [core, startPlugins] = await getStartServices();
+    const { data } = startPlugins;
 
-    // if you need a core dep, we need to pass in more than just getStartServices
+    const sourceConfiguration = await callFetchLogSourceConfigurationAPI(
+      DEFAULT_SOURCE_ID,
+      core.http.fetch
+    );
+
+    const { stats, series } = await fetchLogsOverview(
+      {
+        index: sourceConfiguration.data.configuration.logAlias,
+        timestampField: sourceConfiguration.data.configuration.fields.timestamp,
+      },
+      params,
+      data
+    );
 
-    // perform query
     return {
       title: 'Log rate',
       appLink: 'TBD', // TODO: what format should this be in, relative I assume?
```

Comment:
You can add the relative link here, I will add the basepath from kibana on Observability side.

Reply:
Have we thought through how to apply the same time range for this link?

Reply:
A quick/simple way for now is to make these functions:
```
appLink: ({ startTime, endTime }) => `/app/logs?myStartTimeKey={startTime}&myEndTimeKey={endTime}`
```

Reply:
I will append the query being used in the Observability page to the `appLink`. I think all plugins uses these params, am I right?
`?rangeFrom=now-15m&rangeTo=now&refreshInterval=0&refreshPaused=true`

Reply:
They do not :(

Reply:
Logs and Metrics have a "link-to" helper we can use for now -- we need to fix this globally but for now we may just need to hack around it for this iteration so that these links don't send you to the app without the time range context

@weltenwort do we have docs for "link-to"? 

Not sure if that's easier or if we should just accept a function for each solution's app link

Reply:
So, in that case, I'd go with what you suggested. But we can move forward with this PR. I can change it later.

Reply:
I implemented this in 2ec953feb1b912cc80105c1c0097eaa65679b9dc. The containing function has already the necessary values so there's no need to make another function call.